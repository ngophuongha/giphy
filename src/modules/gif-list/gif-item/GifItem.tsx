import Card from "react-bootstrap/Card";
import React, { Fragment, useCallback, useState } from "react";
import { FullScreen, useFullScreenHandle } from "react-full-screen";
import { GifProps } from "./GifItem.type";
import { GifFullscreen } from "../gif-fullscreen/GifFullscreen";

export const GifItem = ({
  id,
  imageSrc,
  title,
  username,
  rating,
}: GifProps): JSX.Element => {
  const handle = useFullScreenHandle();
  const [show, setShow] = useState(false);
  const toggle = () => {setShow(!show); console.log('show: ', show)};
  const close = () => setShow(false);
  return (
    <Fragment>
      <Card
        style={{ width: "18rem", cursor: "pointer" }}
        id={id}
        className="border-0 overflow-hidden"
        onClick={toggle}
      >
        <img alt="" src={imageSrc} />
        <div className="card-img-overlay h-100 d-flex flex-column justify-content-end">
          <p className="card-text text-white">{username}</p>
        </div>
      </Card>
      <GifFullscreen open={show} setShow={close} imageUrl={imageSrc} /> 
    </Fragment>
  );
};
