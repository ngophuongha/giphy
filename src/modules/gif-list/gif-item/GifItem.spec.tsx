import { render } from "@testing-library/react";
import { GifItem }from "./GifItem";

describe("Loader", () => {
  it("should render without crashing", () => {
    const { container } = render(<GifItem />);
    expect(container).toMatchSnapshot();
  });
});
