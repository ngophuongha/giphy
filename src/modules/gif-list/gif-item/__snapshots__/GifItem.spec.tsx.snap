// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render without crashing 1`] = `
<div>
  <div
    class="border-0 overflow-hidden card"
    style="width: 18rem;"
  >
    <img
      alt=""
    />
    <div
      class="card-img-overlay h-100 d-flex flex-column justify-content-end"
    >
      <p
        class="card-text text-white"
      />
    </div>
  </div>
</div>
`;
